### ISSUE TITLE
Symfony cache is compiled too many times

### ISSUE THREAD

**Issue by sylfabre** [overblog/GraphQLBundle#385]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Version/Branch   | 0.11.5

I don't know how files from `/var/cache/[dev|test|prod]/overblog/graphql-bundle/__definitions__` are loaded but if one of them is not created at the same time as `/var/cache/[dev|test|prod]/src[Dev|Test|Prod]DebugProjectContainer` then when Symfony evaluates if the cache is fresh or not, it wrongly think its not fresh.

Path to the issue:
1. `Symfony\Component\HttpKernel\Kernel::initializeContainer()` calls `$cache->isFresh()`
2. `Symfony\Component\Config::isFresh()` calls its parent method
3. `ResourceCheckerConfigCache::isFresh()` loops through `$meta` on line 93.
The timestamp used as reference is from line 91.
The loop will use `Symfony\Component\Config\Resource\ReflectionClassResource` with one of the class from  `/var/cache/[dev|test|prod]/overblog/graphql-bundle/__definitions__`.
4. `Symfony\Component\Config\Resource\ReflectionClassResource::isFresh()` will then call its method `computeHash` because `$filemtime > $timestamp` is true
5. `new \ReflectionClass($this->className);` will throw an exception and `computeHash()` will return false
=> Symfony thinks that the cache is not fresh

Somehow the classes are loaded but there is another issue with another checker: `\Symfony\Component\Config\Resource\ClassExistenceResource::isFresh()`
1. When Symfony instantiates the class we have `$this->exists === false`
2. On the first call, our definition classes are not loaded so `$loaded = false` and Symfony does not find the class so `$exists = false` so `$this->exists xor !$exists = true`
=> Symfony thinks the cache is fresh
3. On the next calls, our definition classes are not loaded so `$loaded = true` so `$exists = true` so `$this->exists xor !$exists = false`
=> Symfony thinks the cache is not fresh

You can fix the issue while testing with this line in composer.json:
````
"autoload-dev": {
        "psr-4": {
            "App\\Tests\\": "tests/",
            "Overblog\\GraphQLBundle\\__DEFINITIONS__\\": "var/cache/test/overblog/graphql-bundle/__definitions__"
        }
    },
````

I've created a zip file of my minimal code to reproduce the issue:
- download 
[debug-graphql.zip](https://github.com/overblog/GraphQLBundle/files/2398627/debug-graphql.zip)
- do not run composer as I have edited vendor files with `var_dump` to highlight the issue
- run `php bin/phpunit`
=> there is multiple "cache not fresh" messages whereas Symfony cache should stay fresh
- add the line to composer.json
- run `composer dumpautoload`
- run `php bin/phpunit`
=> cache is always fresh and the tests run faster

**Comment by mcg-web** [comment]
Thank for your feedback and the minimal code to reproduce this :+1:. This is very good description of the issue!

**Comment by mcg-web** [comment]
Thanks again for feedback! Fixed in [v0.11.6](https://github.com/overblog/GraphQLBundle/releases/tag/v0.11.6).
